python/tests/transformers/keras_transformer_test.py

```diff
@@ -0,0 +1,145 @@
+# limitations under the License.
+#
+import numpy as np
+import os
+
+from keras.models import Sequential
+from keras.layers import Dense, Activation
+from keras.initializers import glorot_uniform
+
+from sparkdl.transformers.keras_transformer import KerasTransformer
+from ..tests import SparkDLTestCase
+
+class KerasTransformerTest(SparkDLTestCase):
+
+    # TODO(sid): How to create static class variables in Python? Should I just put these in the
+    # constructor?
+    NUM_FEATURES = 10
+    RANDOM_SEED = 997
+
+    def _loadNumpyData(self, num_examples, num_features):
+        """
+        Construct and return a 2D numpy array of shape (num_examples, num_features) corresponding
+        to a dataset of one-dimensional examples.
+        """
+        np.random.seed(self.RANDOM_SEED)
+        return np.random.randn(num_examples, num_features)
+
+    def getInputDF(self, sqlContext, inputCol, idCol):
+        """ Return a DataFrame containing a long ID column and an input column of arrays. """
+        x_train = self._loadNumpyData(num_examples=20, num_features=self.NUM_FEATURES)
+        train_rows = [{idCol : i, inputCol : x_train[i].tolist()} for i in range(len(x_train))]
+        return sqlContext.createDataFrame(train_rows)
+
+    def _getKerasModelWeightInitializer(self):
+        """
+        Get initializer for a set of weights (e.g. the kernel/bias of a single Dense layer)
+        within a Keras model.
+        """
+        return glorot_uniform(seed=self.RANDOM_SEED)
+
+    def getKerasModel(self):
+        """
+        Build and return keras model for (binary) classification on one-dimensional data.
+
+        The model has a single hidden layer that feeds into a single-unit, sigmoid-activated output
+        unit. Weights of the hidden & output layer are drawn from a zero-mean uniform distribution.
+        """
+        model = Sequential()
+
+        # We add a vanilla hidden layer, specifying the initializer (RNG) for the kernel & bias
+        # weights
+        model.add(Dense(units=50, input_shape=(self.NUM_FEATURES, ),
+                        bias_initializer=self._getKerasModelWeightInitializer(),
+                        kernel_initializer=self._getKerasModelWeightInitializer()))
+        model.add(Activation('relu'))
+
+        # We project onto a single unit output layer, and squash it with a sigmoid:
+        model.add(Dense(units=1, bias_initializer=self._getKerasModelWeightInitializer(),
+                        kernel_initializer=self._getKerasModelWeightInitializer()))
+        model.add(Activation('sigmoid'))
+        return model
+
+    def prepareKerasModelFile(self, model, filename):
+        """
+        Saves the passed-in keras model to a temporary directory with the specified filename.
+        """
+        path = os.path.join(self.tempdir, filename)
+        model.save(path)
+        return path
+
+    def executeKerasModel(self, df, model, input_col, id_col):
```

Comment:
maybe `executeKerasModelLocally` is clearer

Reply:
also nit: put below `transformOutputToComparables` so the transformer helper methods are together.